### DISCUSSION TITLE
mapK with method name

### DISCUSSION THREAD

**Discussion by xplosunn** [typelevel/cats-tagless#311] [Q&A]
Currently FunctorK provides mapK:

```scala
def mapK[F[_], G[_]](af: Alg[F])(fk: F ~> G): Alg[G]
```

I've had the use case where I'd like to do a mapK, but have the method name of the Alg (it was a trait) available. A simple example would be me wanting to log every method call on an Alg (using mapK to map to the same effect). 

I think it'd be useful to have something like:

```scala
def namedMapK[F[_], G[_]](af: Alg[F])(methodName: String, fk: F ~> G): Alg[G]
```

If this is something you believe makes sense to have, I'd be happy to contribute. If you believe I'm looking at it the wrong way or there's a better approach, I'll also be happy to read your thoughts.

**Comment by joroKr21** [comment]
What you're looking for is `instrument`: https://github.com/typelevel/cats-tagless/blob/master/core/src/main/scala/cats/tagless/aop/Instrument.scala

**Comment by xplosunn** [comment]
I will look into it. Thanks!
